storage/src/index/storage.rs

```diff
@@ -130,10 +131,9 @@ impl<T: Translator, V: Clone> Index<T, V> {
                 break;
             }
             self.keys_pruned.inc();
-            match head.next {
-                Some(ref mut next) => {
-                    head.value = next.value.clone();
-                    head.next = next.next.take();
+            match head.next.take() {
+                Some(next_box) => {
```

Comment:
nit: I liked calling this `next`

Reply:
lol me too, I will blame the AI again!  reverted.